cpp/src/randomforest/randomforest.cu

```diff
@@ -341,6 +344,100 @@ std::vector<unsigned char> save_model(ModelHandle model) {
+      throw(
+        "Error! the trees saved in concatenated forest and the different "
+        "from the ones present in the individual forests");
+    }
+
+    for (int indiv_trees = 0; indiv_trees < model.trees.size(); indiv_trees++) {
+      const tl::Tree concat_tree =
+        concat_model.trees[concat_mod_tree_num + indiv_trees];
+      const tl::Tree single_worker_tree = model.trees[indiv_trees];
+      if (concat_tree.num_nodes != single_worker_tree.num_nodes) {
+        throw(
+          "Error! the trees saved in concatenated forest and the different "
+          "from the ones present in the individual forests");
+      }
+
+      for (int each_node = 0; each_node < concat_tree.num_nodes; each_node++) {
+        const tl::Tree::Node& node_concat = concat_tree[each_node];
+        const tl::Tree::Node& node_single = single_worker_tree[each_node];
+        bool check_is_root = node_concat.is_root() != node_single.is_root();
+        bool check_parent = node_concat.parent() != node_single.parent();
+        bool check_is_leaf = node_concat.is_leaf() != node_single.is_leaf();
+        bool check_leaf = node_concat.leaf_value() != node_single.leaf_value();
+        bool check_cright = node_concat.cright() != node_single.cright();
+        bool check_cleft = node_concat.cleft() != node_single.cleft();
+        bool check_split =
+          node_concat.split_index() != node_single.split_index();
+
+        if (check_is_root || check_parent || check_is_leaf || check_leaf ||
+            check_split || check_cleft || check_cright) {
+          throw(
+            "Error! the trees saved in concatenated forest and the different "
+            "from the ones present in the individual forests");
+        }
+      }
+    }
+    concat_mod_tree_num = concat_mod_tree_num + model.trees.size();
+  }
+}
+
+std::vector<unsigned char> concatenate_trees(
+  const cumlHandle& handle, std::vector<ModelHandle*> treelite_handles) {
+  tl::Model& first_model = *(tl::Model*)treelite_handles[0];
+  tl::Model concat_model;
+  for (int tl_num = 0; tl_num < treelite_handles.size(); tl_num++) {
+    tl::Model& model = *(tl::Model*)treelite_handles[tl_num];
+    int limit = model.trees.size();
+    for (int i = 0; i < limit; i++) {
+      (concat_model.trees).push_back((model.trees[i]));
+    }
+  }
+
+  ModelHandle concat_mod_handle = &(concat_model);
+  concat_model.num_feature = first_model.num_feature;
+  concat_model.num_output_group = first_model.num_output_group;
+  concat_model.random_forest_flag = first_model.random_forest_flag;
+  concat_model.param = first_model.param;
+  size_t single_handle;
+  TREELITE_CHECK(TreeliteQueryNumTree(concat_mod_handle, &single_handle));
+  TREELITE_CHECK(TreeliteQueryNumFeature(concat_mod_handle, &single_handle));
+  std::vector<unsigned char> concat_mod_bytes;
+  // check if the concatenated model created matches exactly with the original
+  check_concat_tl_mod(concat_mod_handle, treelite_handles);
```

Comment:
Maybe put it behind `#if 0`, so that this check only runs when a developer wants to debug it.

Reply:
I beleive a small check of atleast the basic features should be carried out to check if the concatenated forest information is right. I can create a smaller basic check inside the `check_concat_tl_mod` function to check the toatal number of trees, number of features and output_group of the concatenated forest. The extensive check can be kept as an option if the user wants to debug their code.